### PR TITLE
chore(updatecli/goss): change yaml engine to comply with goss templating on goss-windows and improve goss test

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/4090
the goss exception is failing with a scalar block and updatecli is failing if it's on a one line. To avoid updatecli failure, we need to use the alternate yaml engine `yamlpath`

### CODE REVIEW

**Comment by smerle33** [comment]
even if it seems ok, it's not ... 
```
[2024-05-17T14:18:59.052Z] [0;32m2024-05-17T14:18:58Z:     azure-arm.windows: ......FF...................[0m
[2024-05-17T14:18:59.052Z] [1;31m2024-05-17T14:18:58Z: ==> azure-arm.windows: 2024-05-17T14:18:57Z [DEBUG][Command][visualstudio][stderr]  'C:\Program' is not recognized as an internal or external command,[0m
[2024-05-17T14:18:59.052Z] [0;32m2024-05-17T14:18:58Z:     azure-arm.windows:[0m
[2024-05-17T14:18:59.052Z] [1;31m2024-05-17T14:18:58Z: ==> azure-arm.windows: 2024-05-17T14:18:57Z [DEBUG][Command][visualstudio][stderr]  operable program or batch file.[0m
[2024-05-17T14:18:59.052Z] [1;31m2024-05-17T14:18:58Z: ==> azure-arm.windows: 2024-05-17T14:18:57Z [DEBUG][Command][visualstudio][stderr][0m
[2024-05-17T14:18:59.052Z] [0;32m2024-05-17T14:18:58Z:     azure-arm.windows: Failures/Skipped:[0m
[2024-05-17T14:18:59.052Z] [0;32m2024-05-17T14:18:58Z:     azure-arm.windows:[0m
```

**Comment by smerle33** [comment]
working ! 
windows 2019: `2024-05-21T10:16:02Z: ==&gt; azure-arm.windows: 2024-05-21T10:16:01Z [DEBUG][Command][visualstudio][stdout]  16.11.2.50704`
windows 2022: 
```
12:06:22  
2024-05-21T10:06:19Z:     azure-arm.windows: Command: visualstudio: exit-status: skipped

12:06:22  
2024-05-21T10:06:19Z:     azure-arm.windows: Command: visualstudio: stdout: skipped
```

File: build-jenkins-agent-windows.pkr.hcl [hunk 1]
```diff
@@ -89,14 +89,29 @@ build {
     disable = true
   }
 
+  # goss windows
```
Comment (dduportal) [inline build-jenkins-agent-windows.pkr.hcl:92]:
```suggestion
```

Unused comment (it does not explain anything)

File: build-jenkins-agent-windows.pkr.hcl [hunk 2]
```diff
@@ -89,14 +89,29 @@ build {
...
     inline = [
       "$ErrorActionPreference = 'Stop'",
       "goss --version",
-      "goss --use-alpha=1 --gossfile C:/goss-windows.yaml --loglevel DEBUG validate --retry-timeout 300s",
-      "goss --use-alpha=1 --gossfile C:/goss-common.yaml --loglevel DEBUG validate --retry-timeout 300s",
+      "goss --use-alpha=1 --gossfile C:/goss-windows.yaml --loglevel DEBUG validate --retry-timeout 60s",
+    ]
+  }
+
+  # goss common
```
Comment (dduportal) [inline build-jenkins-agent-windows.pkr.hcl:103]:
```suggestion
```

Same: unused comment

File: build-jenkins-agent-windows.pkr.hcl [hunk 3]
```diff
@@ -89,14 +89,29 @@ build {
...
+  provisioner "powershell" {
+    environment_vars  = local.provisioning_env_vars
+    inline = [
+      "goss --use-alpha=1 --gossfile C:/goss-common.yaml --loglevel DEBUG validate --retry-timeout 60s",
+    ]
+  }
+
+  # cleanup
```
Comment (dduportal) [inline build-jenkins-agent-windows.pkr.hcl:111]:
```suggestion
```

Same

**Review by dduportal — COMMENTED** [review]
Do you mind expanding the 3 comments (to explain the "why") or removing them? In their current state they are not indicating anything for the reader

**Review by dduportal — APPROVED** [review]
🚀

**Pull request merged**